morpheus-graphql-client/test/Case/JSON/Custom/Errors.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Case.JSON.Custom.Errors
+  ( test,
+  )
+where
+
+import Data.ByteString.Lazy.Char8
+  ( ByteString,
+  )
+import Data.Morpheus.Client
+  ( EncodeScalar (..),
+    Fetch (..),
+    ScalarValue (..),
+    gql,
+  )
+import Data.Text (Text)
+import Spec.Utils
+  ( defineClientWithJSON,
+    mockApi,
+  )
+import Test.Tasty
+  ( TestTree,
+  )
+import Test.Tasty.HUnit
+  ( assertEqual,
+    testCase,
+  )
+import Prelude
+  ( Either (..),
+    Eq (..),
+    IO,
+    Show,
+    String,
+    ($),
+  )
+
+newtype GitTimestamp = GitTimestamp
+  { unGitTimestamp :: Text
+  }
+  deriving (Eq, Show)
+
+instance EncodeScalar GitTimestamp where
+  encodeScalar (GitTimestamp x) = String x
+
+defineClientWithJSON
+  "JSON/Custom"
+  [gql|
+    query TestQuery
+      {
+        queryTypeName
+      }
+  |]
+
+resolver :: ByteString -> IO ByteString
+resolver = mockApi "JSON/Custom/Errors"
+
+client :: IO (Either String TestQuery)
+client = fetch resolver ()
+
+test :: TestTree
```

Comment:
now i think that, if we test errors then in ideal case we test both cases regular errors, and partial response